src/IntervalTrees.jl

```diff
@@ -602,7 +598,7 @@ function _push!(t::IntervalBTree{K, V, B},
                 node::InternalNode{K, V, B},
                 entry::V, unique_key::Bool, update::Bool) where {K, V, B}
     i = findidx(node, entry) # key index
-    j = i <= length(node) - 1 && entry >= node.keys[i] ? i + 1 : i # child index
+    j = i <= length(node) - 1 && entry < node.keys[i] ? i : i + 1 # child index
```

Comment:
Please review this code in light of #36. While this is just a workaround a complete elaborate fix will needed as described in #36.